app/models/labware_creators/common_file_handling/csv_file_base.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require './lib/nested_validation'
+require 'csv'
+
+# Part of the Labware creator classes
+module LabwareCreators
+  #
+  # This is an abstract class for handling csv files.
+  #
+  class CommonFileHandling::CsvFileBase
+    include ActiveModel::Validations
+    extend NestedValidation
+
+    validate :correctly_parsed?
```

Comment:
It is not a blocking issue but I think naming of ActiveModel custom validation methods should not suggest they are boolean-returning methods. Their return value is not used. They are supposed to be adding messages to the errors  collection. What makes an instance valid is having an empty errors collection.

Reply:
I also think a utility method to check a condition and return a boolean should not be used as a dual purpose method (as validation method well). It introduces a side effect to the actual purpose of the method.

Reply:
refactored